Document project resources section of contribution guide (#723)
submariner-io/submariner-website: src/content/development/contribution-guide/_index.en.md

## Project Resources

> Submariner uses [GitHub projects](https://docs.github.com/en/issues/organizing-your-work-with-project-boards) to manage a particular release.

Reviewer: ```suggestion
Submariner uses [GitHub Projects](https://docs.github.com/en/issues/organizing-your-work-with-project-boards) to manage releases.
```

> Read [here](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards) to know more on

Reviewer 2: ```suggestion
Read “[Tracking progress on your project board](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards)” to learn
```

It’s better to make link titles explicit.

Reviewer: (for accessibility reasons at least)

> “[Tracking progress on your project board](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards)"

Reviewer: These two quotes are not the same style. I guess that's a non-blocker but I think it would be good to be consistent, at least consistent within the same quote.

```
# The second one, an ASCII double quote
$ echo "\"" | tr -d "\n" | od -An -t dC
   34
# The first one, I guess a unicode double quote?
$  echo "“" | tr -d "\n" | od -An -t dC
  -30 -128 -100
```

Reviewer 3: ascii one makes more sense IMHO

> “[Tracking progress on your project board](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards)"

Reviewer 3: Please put long links in the end of the doc as it makes the source text much easier to read and follow.

Author: I put all links in the end for easy readability and consistency.

how to work with projects.

### Backlog Board

> [Backlog board](https://github.com/orgs/submariner-io/projects/15) is maintained to host any issues or features that would not
> land in the current release.

Reviewer 2: ```suggestion
[The backlog board](https://github.com/orgs/submariner-io/projects/15) hosts issues or features that are not scheduled
for the release currently being developed.
```

> for the release currently being developed.

Reviewer 3: Please add some explanation on:
* how to work with the board in general (e.g. make sure to assign any task/epic to the backlog)
* general details on the board - e.g. "while tasks and epics are reported in their corresponding repositories, this board given an aggregated view on all planned work"

Do we even want to track bugs in the backlog?
I think for bugs it just adds extra noise and has no value.
If so, we should specify that only tasks/epics are expected to be tracked on the backlog.

Reviewer 4: @mkolesnik where do you think is best place to track bugs?
Since a project board is giving aggregated view from all our repositories, it make sense to have the bugs tracked on either of our project boards (backlog or current version)

Author: I think we should track any issue open in backlog board.

Reviewer 3: We can track bugs with queries (as we do today), I don't see a particular reason to add them to backlog but if it makes sense to everyone else I won't object.

### Current Release Board

> Current release work is tracked at [0.13 board](https://github.com/orgs/submariner-io/projects/20)

Reviewer 2: ```suggestion
Current release work is tracked in [the 0.13 board](https://github.com/orgs/submariner-io/projects/20).
```

> Current release work is tracked in [the 0.13 board](https://github.com/orgs/submariner-io/projects/20).

Reviewer 3: I think it would be good to document how the board is structured and the expected workflow (e.g. move to in review when a PR is created, move to verification when its merged, etc).

> Current release work is tracked in [the 0.13 board](https://github.com/orgs/submariner-io/projects/20).

Reviewer 3: Do we want to link directly to the 0.13 board? We will have to remember to update it each version :)

Author: What else can we do?

Reviewer 3: Just link to the projects page?

### Enhancements Repository

All enhancement proposals need to be submitted to the [enhancements repository](https://github.com/submariner-io/enhancements). To submit a
new enhancement proposal, raise an 
[Enhancement Request issue](https://github.com/submariner-io/enhancements/issues/new?assignees=&labels=enhancement&template=enhancement.md)

> on the repository.

Reviewer 3: From this phrasing it sounds like any enhancement to submariner should ive here. While this may be useful for unaware users, we should probably move them (or urge to open) to the specific repositories for small/medium tasks.

I'd also document the different usage we have here, e.g. the "issues" are used to propose and track epics, the "pull requests" are used to discuss design proposals for implementing the epics.

Author: I think that should be explained under `issues` and `epics` section.

### Releases Repository

> Submariner release is automated to a great extend. This [section](https://submariner.io/development/release-process/) explains the details

Reviewer 2: ```suggestion
Submariner release is automated to a great extent. [The release process documentation](https://submariner.io/development/release-process/) explains the details
```

Reviewer: That plus: `Submariner's release` or `Submariner's releases are`

> Submariner's release is automated to a great extent. [The release process documentation](https://submariner.io/development/release-process/)

Reviewer 3: ```suggestion
Submariner's release is automated to a great extent. The [release process documentation](https://submariner.io/development/release-process/)
```

Also please put long links at end of file.

> of a release. The code is maintained in the [releases repository](https://github.com/submariner-io/releases)

Reviewer 2: ```suggestion
of a release. The release automation is maintained in the [releases repository](https://github.com/submariner-io/releases).
```

> explains the details of a release. The release automation is maintained in the [releases repository](https://github.com/submariner-io/releases).

Reviewer 3: You should add details that all submariner releases (containing the subctl binaries) can be found here.

## Project Resources

> Submariner uses [GitHub Projects] to manage releases. Read [Tracking progress on your project board] to learn more on how to work with 

Reviewer: ```suggestion
Submariner uses [GitHub Projects] to manage releases. Read [Tracking Progress on Your Project Board] to learn more on how to work with 
```

I know the GitHub folks are trying to look modern with their no-cap titles, but it doesn't look like an article title if we don't capitalize it properly inline.

Per https://capitalizemytitle.com

Reviewer 2: > I know the GitHub folks are trying to look modern with their no-cap titles, but it doesn't look like an article title if we don't capitalize it properly inline.

That’s a matter of taste ;-). But we are mostly using US-style title capitalisation, so it makes sense to use it throughout.

projects.

### Backlog Board

> [The backlog board] hosts issues or features that are intended to be worked upon. While each issue is opened in

Reviewer: ```suggestion
[The backlog board] hosts issues or features that are intended to be worked on eventually. While each issue is opened in
```

This is better grammer, but I don't know if the content makes sense. I think all open Issues are intended to be worked on, not just ones on the backlog.

Maybe something like

```suggestion
[The backlog board] hosts issues or features that are not a part of a particular release board. While each issue is opened in
```

Author: > This is better grammer, but I don't know if the content makes sense. I think all open Issues are intended to be worked on, not just ones on the backlog.
> 
> Maybe something like

All issues opened as supposed to be in `Backlog` board.

Reviewer 3: I think Backlog is a convenient place to put all enhancements and it allows us to triage and prioritize them.
We could also include bugs there although that would be more tedious but maybe it would be beneficial to have them all in one place?

It's not that issues outside of backlog aren't going to be worked on, it's just that theyre not necessarily triaged and thus less likely to be worked on.

Reviewer: Right, that's what I'm trying to make clear with the second option.

> [The backlog board] hosts issues or features that are not scheduled for the release currently being developed. While each issue is opened in

Reviewer: ```suggestion
[The backlog board] hosts issues or features that are not scheduled for the release currently being developed. While each issue is opened in
```

This is not true anymore, right? We're now keeping things on the backlog board, with `next-version-candidate` and prio labels, and letting devs move them to 0.13 I think?

Reviewer 3: Yes this content needs to be updated that any small/medium tasks are kept in this board and moved to the project board when devs pick them up, which is done in the order or priority.

its corresponding repository, this board gives an aggregated view of all open issues across all repositories. The board has total of seven
columns. Any issue or epic opened should be assigned to `Backlog` Projects in `Backlog` column. Issues intenteded to be

> worked on for the next release are moved to `Next Version Candidate` column. During [pre-planning](#Pre-Planning), issues from the `Next

Reviewer 3: I thought we work on issues by priority?

Version Candidate` column are triaged. Issues or epics properly triaged, with assignees and priority labels addeded, are moved to [current

> release board](#Current Release Board). The next 3 columns are of priority. All unassigned issues are sorted into these three columns

Reviewer: ```suggestion
release board](#Current Release Board). The next 3 columns sort by priority. All unassigned issues are sorted into these three columns
```

Author: `sort by` doesn't make sense. I intend to say that the next columns `are for` priority.

Reviewer: The three columns in question are sorted by priority, so they "sort by priority".

> based on the priority discussed in the team meeting. Issues from these columns should be worked upon if the current release work is

Reviewer: ```suggestion
based on the priority discussed in the team meeting. Issues from these columns should be worked on if the current release work is
```

finished. Any leftover work from the previous release goes to the `Work in Progess column`. `Close?` column holds any issue/epic that
could probably be closed for various reasons.

### Current Release Board

> Current release work is tracked in [the 0.13 board]. THe board has 5 columns. The first one, `Schedule and Epics`, hosts all properly

Reviewer: ```suggestion
Current release work is tracked in [the 0.13 board]. Tee board has 5 columns. The first one, `Schedule and Epics`, hosts all properly
```

Do we really want to mention a specific version? That's going to be out of date very soon and require maintenance.

Author: How else to link the "current" board?

Reviewer: I'd just link to https://github.com/orgs/submariner-io/projects and let the user select the current one.

Reviewer: I see now this is the same feedback Mike gave elsewhere, with the same fix proposed.

> triaged epics targeted for the current release and schedule for the current release. Triaged issues are under `TO DO` column. When an

Reviewer: ```suggestion
triaged epics targeted for the current release and the schedule for the current release. Triaged issues are under `TO DO` column. When an
```

> triaged epics targeted for the current release and schedule for the current release. Triaged issues are under `TO DO` column. When an

Reviewer: ```suggestion
triaged epics targeted for the current release and schedule for the current release. Triaged issues are under the `TODO` column. When an
```

Author: The column is named "To do" with the space. I changed the case.

issue is being worked on, it is moved to `In Progress` column. When a PR for the issue is pushed, the issue is moved to `In Review`

> column. The PR has the corresponding issue linked from the GitHub UI and is not tracked on the board. Once the PR is merged, the issue

Reviewer: > corresponding issue linked from the GitHub UI and is not tracked on the board

I'm not really a fan of this practice because it reduces the visibility of PRs, which we very much need more visibility and review of.

Author: Agree. Also linking both issues and PRs to the board adds more noise. The right/preferred/recommended practise is to first open an issue, add it to the board and link the PR to the issue. The board shows the PRs linked to the issue inthe issue card.
